networkx/generators/tests/test_random_graphs.py

```diff
@@ -137,28 +142,49 @@ def test_dual_barabasi_albert(self, m1=1, m2=4, p=0.5):
         The graphs generation are repeated several times to prevent lucky shots
 
         """
-        seed = 42
-        repeats = 2
+        seeds = [42, 314, 2718]
+        initial = nx.complete_graph(10)
 
-        while repeats:
-            repeats -= 1
+        for seed in seeds:
 
             # This should be BA with m = m1
             BA1 = barabasi_albert_graph(100, m1, seed)
             DBA1 = dual_barabasi_albert_graph(100, m1, m2, 1, seed)
-            assert BA1.size() == DBA1.size()
+            assert BA1.edges() == DBA1.edges()
 
             # This should be BA with m = m2
             BA2 = barabasi_albert_graph(100, m2, seed)
             DBA2 = dual_barabasi_albert_graph(100, m1, m2, 0, seed)
-            assert BA2.size() == DBA2.size()
+            assert BA2.edges() == DBA2.edges()
+
+            BA3 = barabasi_albert_graph(100, m1, seed)
+            DBA3 = dual_barabasi_albert_graph(100, m1, m1, p, seed)
+            # We can't compare edges here since randomness is "consumed" when drawing
+            # between m1 and m2
+            assert BA3.size() == DBA3.size()
+
+            DBA = dual_barabasi_albert_graph(100, m1, m2, p, seed, initial=initial)
+            BA1 = barabasi_albert_graph(100, m1, seed, initial=initial)
+            BA2 = barabasi_albert_graph(100, m1, seed, initial=initial)
+            assert (
+                min(BA1.size(), BA2.size()) <= DBA.size() <= max(BA1.size(), BA2.size())
+            )
 
         # Testing exceptions
         dbag = dual_barabasi_albert_graph
         pytest.raises(NetworkXError, dbag, m1, m1, m2, 0)
         pytest.raises(NetworkXError, dbag, m2, m1, m2, 0)
         pytest.raises(NetworkXError, dbag, 100, m1, m2, -0.5)
         pytest.raises(NetworkXError, dbag, 100, m1, m2, 1.5)
+        pytest.raises(
+            NetworkXError,
+            dbag,
+            100,
+            m1,
+            m2,
+            p,
+            initial=nx.complete_graph(max(m1, m2) - 1),
+        )
```

Comment:
This might make it easier to see that p has changed as well as initial relative to previous lines. 
```suggestion
        initial=nx.complete_graph(max(m1, m2) - 1)
        pytest.raises(NetworkXError, dbag, 100, m1, m2, p, initial=initial)
```

Reply:
Agreed, fixed it !